### PR TITLE
added the sauce function

### PR DESCRIPTION
**Pull request by lburns660**

### CODE REVIEW

File: src/BuildASub.java [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
-        System.out.println("Your chosen meat is: " + meat);
-    }
+        public static void main(String[] args) throws Exception {
+            String bread = chooseBread();
+            String meat = chooseMeat();
+            String sauce = chooseSauce();
+
+            System.out.println("Your chosen bread is: " + bread);
+            System.out.println("Your chosen meat is: " + meat);
+            System.out.println("Your chosen sauce is:" + sauce);
```
Comment (John-Doherty01) [inline src/BuildASub.java]: Calum and Maria are handling outputs, please remove sauce output line

File: src/BuildASub.java [hunk 2]
```diff
@@ -44,20 +46,55 @@ public static String chooseMeat() throws Exception {
         int meatOption = scan.nextInt();
```
Comment (John-Doherty01) [inline src/BuildASub.java:62]: Remove excess spacing

File: src/BuildASub.java [hunk 3]
```diff
@@ -44,20 +46,55 @@ public static String chooseMeat() throws Exception {
...
+    }
+
+        public static String chooseSauce() {
+        String sauceType = "";
+        System.out.print("Would you like any sauce on your sub?");
+
+        Scanner scan = new Scanner(System.in);
+        String sauceOption = scan.nextLine();
+
+        if (sauceOption.equalsIgnoreCase("yes")) {
```
Comment (John-Doherty01) [inline src/BuildASub.java:89]: Allow for letters y/n rather than yes/no

File: src/BuildASub.java [hunk 4]
```diff
@@ -44,20 +46,55 @@ public static String chooseMeat() throws Exception {
...
+
+            }
+        }
+        else if (sauceOption.equalsIgnoreCase("none")){
+                sauceType = "no sauce required";
         }
         else {
-            throw new Exception("Meat not set!");
+            System.out.print("ERROR: please try again \n");
+                    chooseSauce();
```
Comment (John-Doherty01) [inline src/BuildASub.java:109]: Other possibility instead of recursion is the use of exceptions

**Review by John-Doherty01 — CHANGES_REQUESTED** [review]

File: src/BuildASub.java (hunk 2 above)
Comment (lburns660) [reply src/BuildASub.java:62]: hey John hook me up

**Review by John-Doherty01 — APPROVED** [review]

**Pull request merged**